### PR TITLE
[WiP] Phantom JS

### PR DESCRIPTION
**Pull request by mistercrunch**
Did some work to get a headless browser going last april but never got to a perfect state. I'd say this is 60-80% done, needs a bit more TLC

### CODE REVIEW

**Comment by masterlittle** [comment]
Can this be extended to capture screenshots of dashboard and shooting a mail at some intervals?

File: superset/views/images.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        """
+        url = request.url_root[:-1] + path
+        print('-='*80)
+        print(url)
+        self.driver.get(url)
+
+        # The non-zero window height is merely a lower bound.
+        self.driver.set_window_size(width, height)
+        wait = ui.WebDriverWait(self.driver, 30)
+        wait.until(EC.presence_of_element_located(
```
Comment (john-bodley) [inline superset/views/images.py]: We used this logic which works for slices or dashboards,
```
WebDriverWait(self.driver, 1).until_not(
    EC.presence_of_all_elements_located(
        (By.CLASS_NAME, 'loading'),
    ),
)
```

File: superset/views/images.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        window height is resized if needed to ensure that the entire page is
+        encapsulated within the window.
+        :param path: The Superset URL path
+        :type path: str
+        :param wait: The maximum time to wait for AJAX content in seconds
+        :type wait: int
+        :param width: The screenshot width in pixels
+        :type width: int
+        :returns: The captured image
+        :rtype: PIL.Image
```
Comment (john-bodley) [inline superset/views/images.py]: Nit. io.BytesIO

File: superset/views/images.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        :raises HTTPException: If the HTTP request fails
+        :raises socket.timeout: If the page becomes non-responsive
```
Comment (john-bodley) [inline superset/views/images.py]: Nit. Remove

**Comment by mistercrunch** [comment]
I opened this PR as a reference in case others want to pick it up. I got it to a semi-working state and never finished the work.

**Pull request closed without merging**